connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -497,12 +506,18 @@ private SinkRecord convertAndTransformRecord(final ConsumerRecord<byte[], byte[]
                 timestamp,
                 msg.timestampType(),
                 headers);
+
+        InternalSinkRecord internalSinkRecord = origRecord.newRecord(origRecord.topic(),
+            origRecord.kafkaPartition(), origRecord.keySchema(), origRecord.key(),
+            origRecord.valueSchema(), origRecord.value(), origRecord.kafkaOffset(),
+            origRecord.timestamp(), origRecord.timestampType(), origRecord.headers(), msg);
+
         log.trace("{} Applying transformations to record in topic '{}' partition {} at offset {} and timestamp {} with key {} and value {}",
                 this, msg.topic(), msg.partition(), msg.offset(), timestamp, keyAndSchema.value(), valueAndSchema.value());
         if (isTopicTrackingEnabled) {
-            recordActiveTopic(origRecord.topic());
+            recordActiveTopic(internalSinkRecord.topic());
         }
-        return transformationChain.apply(origRecord);
+        return transformationChain.apply(internalSinkRecord);
```

Comment:
This line would change to:
```suggestion
        // Apply the transformations
        SinkRecord transformedRecord = transformationChain.apply(sinkRecord);
        if (transformedRecord == null) {
            // The record is being dropped
            return null;
        }
        // Error reporting will need to correlate each sink record with the original consumer record
        return new InternalSinkRecord(msg, transformedRecord);
```